File: apps/backend/src/main/kotlin/com/lyra/app/organization/domain/OrganizationId.kt
package com.lyra.app.organization.domain

import com.lyra.common.domain.BaseId
import java.util.*

data class OrganizationId(private val id: UUID) : BaseId<UUID>(id) {
    constructor(id: String) : this(UUID.fromString(id))
    companion object {
        fun create() = OrganizationId(UUID.randomUUID())
    }
}

File: apps/frontend/apps/lyra-app/domain/organization/OrganizationId.ts
import { UUIDValueObject } from '@lyra/shared';

export default class OrganizationId extends UUIDValueObject {
  private constructor(value: string) {
    super(value);
    this.ensureIsValidUuid(value);
  }

  static random(): OrganizationId {
    return new OrganizationId(crypto.randomUUID());
  }

  public static create(value: string): OrganizationId {
    return new OrganizationId(value);
  }
}
